docs/operations/metrics.md

```diff
@@ -196,8 +196,9 @@ task's `IOConfig` as follows:
 |`false`|`false`|`REPLACE_LEGACY`. The default for JSON-based batch ingestion. |
 |`false`|`true`|`REPLACE`|
 
-The `tags` dimension is reported only for metrics emitted from ingestion tasks whose ingest spec specifies the `tags`
-field in the `context` field of the ingestion spec. `tags` is expected to be a map of string to object.
+The `tags` dimension enhances metrics, task reports, and the Peon service/heartbeat metric associated with ingestion tasks. These tags are derived from both the ingestion specification—when the `tags` field is specified within the task's context—and system-generated sources. The tags field within a task's `context` should be a map with string keys and object values, facilitating the inclusion of custom metadata. However, tags can also be automatically generated and added by the system, providing a comprehensive set of metadata for monitoring and analysis purposes.
+
+To further customize and enrich task metadata, developers can implement the `TaskContextEnricher` interface. By implementing custom logic within this interface, additional context fields can be introduced. This capability allows for enhanced observability and management of tasks, as these additional context fields can offer deeper insights into task execution, performance, and outcomes.
```

Comment:
I don't think the changes are needed to this part of the doc. Lower down in the doc, the dimensions that are expected with each metric are called out explicitly - since this patch just updates the service/heartbeat metric, can you update that [part of the doc](https://github.com/apache/druid/blob/master/docs/operations/metrics.md#service-health) with this new dimension.
```suggestion
The `tags` dimension is reported only for metrics emitted from ingestion tasks whose ingest spec specifies the `tags`
field in the `context` field of the ingestion spec. `tags` is expected to be a map of string to object.
```